tests/std/tests/P2321R2_views_zip_transform/env.lst

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+RUNALL_INCLUDE ..\strict_concepts_latest_matrix.lst

```

Comment:
### **tl;dr:** Yes, the `strict` requirement is necessary, and I'll be adding a comment explaining why.

It seems like even for `zip_transform_view`, the ADL overload of `iter_move()` for `zip_view::_Iterator` prevents the test from compiling without `/permissive-`. Yes, it's true that `zip_transform_view::_Iterator` doesn't have its own overloads of `iter_move()` and/or `iter_swap()`. However, it seems like somewhere down the line of template instantiations for `zip_transform_view`, the implementation of the aforementioned `iter_move()` overload triggers a compilation failure with `/permissive`.

Looking at the output generated when compiling with `/permissive`, it seems that the error is caused by the following chain of template instantiations:

-> The `range<const _Inner_view>` requirement for `zip_transform_view::begin() const` and `zip_transform_view::end() const` (where `_Inner_view` is the `zip_view` type used internally by the `zip_transform_view`)
    -> The `_Begin::_Cpo::_Choose()` function calling `zip_view::begin()` since `_Has_member` is true, resulting in `zip_view::_Iterator` being instantiated (This occurs because the `range` concept is defined in terms of `ranges::begin()`.)
        -> The instantiation of the `iter_move()` overload for `zip_view::_Iterator`, whose implementation uses `iter_move()` as a parameter to the `_Tuple_transform()` function and thus triggers the ambiguous overload compilation error
        
Given this, it seems very likely that most (if not all) range types implemented in terms of `zip_view` are going to need the `strict` requirement for their associated tests. 

(As a side note, I tried removing the `iter_move()` overload from `zip_view::_Iterator` and compiling the test with `/permissive`. After renaming some unfortunately shadowed identifiers and replacing some `requires` expressions in `zip_transform_view::_Iterator` with `constexpr bool` variables to work around a false C7608 error, the test *did* compile and run with `/permissive`. This obviously isn't a solution to the problem, since we need that overload to be conformant; I'm just confirming the root cause.)

Reply:
Thanks for looking into this!